### PR TITLE
feat(config): add set_extra_header method

### PR DESCRIPTION
**Pull request by chloelbn**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no

### CODE REVIEW

**Review by DevinCodes — CHANGES_REQUESTED** [review]
Looks good to me! However, I think there's a couple of things we need to do before this is ready to merge:
* We need to update the `.rbs` definition of `algolia_config` so that this new method is included.
* We need to update the upgrade guide in the root of this repository.
* We need to update the documentation. This has to be done in 2 locations: the upgrade guide, and the [`set_extra_header`](https://www.algolia.com/doc/api-reference/api-methods/set-extra-header/) page. I've changed the `set_extra_header` page to reflect setting the headers using `config.headers['z'] = 'y'`, but we need to document this method now.

Let me know when the first 2 points are done, and I'll approve the PR 😄

**Comment by chloelbn** [comment]
@DevinCodes changes are pushed!

File: upgrade_guide.md [hunk 1]
```diff
@@ -65,6 +65,20 @@ index.search('query', opts)
 
 ### `Client`
 
+#### `set_extra_header`
+The `set_extra_header` method has been moved from the Client to the `Algolia::AlgoliaConfig` class. You have to define your extra headers on Client instantiation.
```
Comment (DevinCodes) [inline upgrade_guide.md:69]:
```suggestion
The `set_extra_header` method has been moved to the `Algolia::AlgoliaConfig` class. You have to define your extra headers before you instantiate the `Client`.
```

File: upgrade_guide.md [hunk 2]
```diff
@@ -65,6 +65,20 @@ index.search('query', opts)
...
+```ruby
+# Before
+client.set_extra_header('admin', 'admin-key')
+
+# After
+# `Algolia::Search::Config` inherits from `Algolia::AlgoliaConfig`
```
Comment (DevinCodes) [inline upgrade_guide.md:75]:
```suggestion
# Algolia::Search::Config inherits from Algolia::AlgoliaConfig
```

**Review by DevinCodes — CHANGES_REQUESTED** [review]

**Review by DevinCodes — APPROVED** [review]

**Pull request merged**